build/Dockerfile

```diff
@@ -110,29 +110,31 @@ RUN dnf install -y git && \
     rm -f /usr/share/doc/perl-Net-SSLeay/examples/server_key.pem && \
     dnf clean all
 USER ${USER}
-WORKDIR /tmp
+# Ensure that git directory is owned by current user, otherwise git raises
+# "fatal: detected dubious ownership" for `/tmp`
+WORKDIR /tmp/fms-hf-tuning
+
 RUN --mount=type=cache,target=/home/${USER}/.cache/pip,uid=${USER_UID} \
-    python -m pip install --user build
-COPY --chown=${USER}:root tuning tuning
-COPY .git .git
-COPY pyproject.toml pyproject.toml
+    python -m pip install --user wheel packaging poetry poetry-plugin-export
```

Comment:
I don't think we want the poetry and its plugins installed into the /home/tuning/.local/lib/python3.11/site-packages and copied to the final release stage because it's for development.


Maybe remove `build` packages because is not used anymore.
Also the `packaging` is already listed as dependencies the pyproject.toml did you have a reason to install here?

Reply:
Thanks, that's a good find!

While fixing this I noticed that I had made another mistake. I was using poetry `groups` instead of poetry `extras`. However, poetry groups are incompatible with pip i.e. `pip install fms-hf-tuning[flash-attn]` wouldn't install the packages from the `flash-attn` group (which in this example would resolve to just the `flash-attn` python package).

I fixed `pyproject.toml`, the related section in `CONTRIBUTING.md`, and the Dockerfile. I also tested building the dockerfile locally with and without using the `WHEEL_VERSION` build-arg and it seems to be working fine.